examples/plotting/file/custom_datetime_axis.py

```diff
@@ -0,0 +1,58 @@
+from math import pi
+
+import pandas as pd
+
+from bokeh.sampledata.stocks import MSFT
+from bokeh.plotting import figure, show, output_file
+from bokeh.models.formatters import TickFormatter, String, List
+
+class DateGapTickFormatter(TickFormatter):
+  date_labels = List(item_type=String)
+  __implementation__ = """
+_ = require "underscore"
+HasProperties = require "common/has_properties"
+
+class DateGapTickFormatter extends HasProperties
+  type: 'DateGapTickFormatter'
+
+  format: (ticks) ->
+    date_labels = @get("date_labels")
+    labels = ( 
+      for tick, i in ticks
+        date_labels[tick] ? ""
+      )
+    return labels
+
+module.exports =
+  Model: DateGapTickFormatter
+		"""
+
+df = pd.DataFrame(MSFT)[:50]
+df = df.set_index([range(0,len(df))])
```

Comment:
Yup this is causing the one remaining test failure: https://travis-ci.org/bokeh/bokeh/jobs/96149634#L1032